io/net/infiniband/ib_bw_perf.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Prudhvi Miryala<mprudhvi@linux.vnet.ibm.com>
+# Bandwidth Performance test
+# ib_send_bw test
+# ib_write_bw test
+# ib_read_bw test
+# ib_atomic_bw test
+
+
+import time
+import netifaces
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado.utils import distro
+
+
+class Bw_Perf(Test):
+    '''
+    check the availability of perftest package installed
+    perftest package should be installed
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        detect_distro = distro.detect()
+        depends = ["openssh-clients"]
+        if detect_distro.name == "Ubuntu" or detect_distro.name == "redhat":
+            depends.append("perftest")
+        for package in depends:
+            if not sm.check_installed(package):
+                self.skip("%s package is need to test" % package)
+        interfaces = netifaces.interfaces()
+        self.flag = self.params.get("ext_flag", default="0")
+        self.IF = self.params.get("Iface", default="")
+        self.PEER_IP = self.params.get("PEERIP", default="")
+        if self.IF not in interfaces:
+            self.skip("%s interface is not available" % self.IF)
+        if self.PEER_IP == "":
+            self.skip("%s peer machine is not available" % self.PEER_IP)
+        self.CA = self.params.get("CA_NAME", default="mlx4_0")
+        self.PORT = int(self.params.get("PORT_NUM", default="1"))
+        self.PEER_CA = self.params.get("PEERCA", default="mlx4_0")
+        self.PEER_PORT = int(self.params.get("PEERPORT", default="1"))
+        self.to = self.params.get("timeout", default="600")
+
+    def bwperf_exec(self, arg1, arg2, arg3):
+        '''
+        bandwidth performance exec function
+        '''
+        logs = "> /tmp/ib_log 2>&1 &"
+        msg = "timeout %s %s -d %s -i %d %s %s %s" \
+            % (self.to, arg1, self.PEER_CA, self.PEER_PORT,
+                arg2, arg3, logs)
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("ssh failed to remote machine")
```

Comment:
is this indicate only ssh connection setup failed or run of the command in Remote system also fail ?